packages/terra-doc-template/src/DocTemplate.jsx

```diff
@@ -10,13 +10,28 @@ const cx = classNames.bind(styles);
 
 const propTypes = {
   /**
-   * The given component's npm package name
+   * The given component's npm package name.
    */
   packageName: PropTypes.string,
   /**
-   * The given component's readme file imported to a string
+   * The given component's readme file imported to a string.
    */
   readme: PropTypes.string,
+  /**
+   * The path from the project root to the folder for the component. Only needs to be specified
+   * if `packages/${packageName}` would be wrong.
+   */
+  srcPath: PropTypes.string,
+  /**
+   * The URL for the git repository of the project. Typically found by importing `{ repository }` from
+   * the `package.json` and passing the `repository.url`. This prop and either `packageName` or `srcLink`
+   * are required to be specified for the source link to be displayed on the page.
+   */
+  repositoryURL: PropTypes.string,
```

Comment:
This will be cleaner if we remove the `repositoryURL` and `branch` props and the associated logic checks with it and require consumers to just provide the direct URL via the `srcPath` prop.